runtime/compiler/ilgen/Walker.cpp

```diff
@@ -3526,8 +3578,30 @@ TR_J9ByteCodeIlGenerator::genInvoke(TR::SymbolReference * symRef, TR::Node *indi
    int32_t numArgs = calledMethod->numberOfExplicitParameters() + (isStatic ? 0 : 1);
 
    TR::ILOpCodes opcode = TR::BadILOp;
-   switch (symbol->getRecognizedMethod())
-      {
+   switch (auto method = symbol->getRecognizedMethod())
+      {
+      // Eliminate calls to unboxing methods if they can be replaced with a load of the
+      // `value` field. If the `value` field symbol reference cannot be found, e.g.
+      // The underlying implementation of the boxed class has changed, emit the call
+      // to getter as normal.
+      case TR::java_lang_Boolean_booleanValue:
+      case TR::java_lang_Byte_byteValue:
+      case TR::java_lang_Character_charValue:
+      case TR::java_lang_Short_shortValue:
+      case TR::java_lang_Integer_intValue:
+      case TR::java_lang_Long_longValue:
+      case TR::java_lang_Float_floatValue:
+      case TR::java_lang_Double_doubleValue:
+         if (comp()->getOption(TR_DisableIntrinsics))
+            break;
```

Comment:
I notice that `TR_DisableIntrinsics` was originally implemented to bundle another pair of options related to packed decimal intrinsic operations:  [TR_DisableMarshallingIntrinsics](https://github.com/eclipse/omr/blob/7bf073fd4538feb566c4a3175c3d460cc59d8114/compiler/control/OMROptions.cpp#L443) and [TR_DisablePackedDecimalIntrinsics](https://github.com/eclipse/omr/blob/7bf073fd4538feb566c4a3175c3d460cc59d8114/compiler/control/OMROptions.cpp#L489).  See [J9Options.cpp](https://github.com/eclipse-openj9/openj9/blob/8ea33906c53711d12220d3cc88c10dfcb97fd71a/runtime/compiler/control/J9Options.cpp#L3242-L3252).  I wonder whether this use deserves its own option.

Another possibility might be to change the existing `TR_DisableIntrincis` option from a toggle to a pattern matching glob (so called, regex) option.  That would make it easier to control recognition of specific intrinsics.  That would be a bigger task, and could be done as a follow on item.

A third possibility might be to completely eliminate `TR_DisableMarshallingIntrinsics` and `TR_DisablePackedDecimalIntrinsics`, forcing all intrinsics to be turned off by `TR_DisableIntrinsics`.

Reply:
I think for the short term, we can eliminate `TR_DisableMarshallingIntrinsics` and `TR_DisablePackedDecimalIntrinsics` and disable all intrinsics via the `TR_DisableIntrinsics`. I can open a PR concurrently which will remove those individual options and delegate those intrinsics to `TR_DisableIntrinsics`. 

In the longer term, it would be nice to have the matching glob functionality for the `TR_DisableIntrinsics`. I can open an issue for `> 0.44`.

I think individual options would have too much granularity as there are many intrinsic candidates to maintain individual options for and it doesn't make sense to have options for some, but not all the candidates.